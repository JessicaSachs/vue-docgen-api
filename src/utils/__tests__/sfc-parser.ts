import { CompiledSFC } from 'vue-sfc';
import scfParser from '../sfc-parser';

describe('sfc-parser', () => {
  let compiledSfc: CompiledSFC;
  beforeEach(() => {
    compiledSfc = scfParser(
      `
    <template lang="pug"><div/></template>
    <script lang="ts">
    export default {}
    </script>
    <style lang="scss">.test{display:block;}</style>
    `,
      'test',
    );
  });

  it('should parse sfc templates', () => {
    expect(compiledSfc).toMatchSnapshot();
  });
});
