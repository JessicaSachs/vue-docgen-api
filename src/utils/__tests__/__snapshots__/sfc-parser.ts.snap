// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sfc-parser should parse sfc templates 1`] = `
Object {
  "customBlocks": Array [],
  "script": Object {
    "attrs": Object {
      "lang": "ts",
    },
    "content": "


export default {}
",
    "end": 93,
    "lang": "ts",
    "start": 66,
    "type": "script",
  },
  "styles": Array [
    Object {
      "attrs": Object {
        "lang": "scss",
      },
      "content": "




.test{display:block;}",
      "end": 147,
      "lang": "scss",
      "start": 126,
      "type": "style",
    },
  ],
  "template": Object {
    "attrs": Object {
      "lang": "pug",
    },
    "content": "<div/>",
    "end": 32,
    "lang": "pug",
    "start": 26,
    "type": "template",
  },
}
`;
